server/routerlicious/packages/lambdas/src/deli/lambda.ts

```diff
@@ -320,6 +336,52 @@ export class DeliLambda extends EventEmitter implements IPartitionLambda {
         this.clearOpMaxTimeTimer();
 
         this.removeAllListeners();
+
+        this.logSessionEndMetrics(closeType);
+    }
+
+    private logSessionStartMetrics() {
+        if (this.sessionStartMetric?.isCompleted()) {
+            this.sessionStartMetric = this.createSessionMetric();
+        }
+
+        if (this.isNewDocument) {
+            if (this.scribeStartSuccess) {
+                this.sessionStartMetric?.setProperties({ [CommonProperties.sessionState]: SessionState.started });
+                this.sessionStartMetric?.success("Session started successfully");
+            } else {
+                this.sessionStartMetric?.setProperties({ [CommonProperties.sessionState]:
+                    SessionState.scribeLambdaDown });
+                this.sessionStartMetric?.error("Scribe lambda failed");
+            }
+        } else {
+            this.sessionStartMetric?.setProperties({ [CommonProperties.sessionState]: SessionState.resumed });
+            this.sessionStartMetric?.success("Session resumed successfully");
+        }
+    }
+
+    private logSessionEndMetrics(closeType: LambdaCloseType) {
+        if (this.sessionMetric?.isCompleted()) {
+            this.sessionMetric = this.createSessionMetric();
+        }
+
+        this.sessionMetric?.setProperties({ [CommonProperties.sessionEndReason]: closeType });
+        this.sessionMetric?.setProperties({ [CommonProperties.sequenceNumber]: this.sequenceNumber });
+        this.sessionMetric?.setProperties({ [CommonProperties.lastSummarySequenceNumber]: this.durableSequenceNumber });
+
+        if (closeType === LambdaCloseType.Error) {
+            this.sessionMetric?.setProperties({ [CommonProperties.sessionState]: SessionState.end });
+            this.sessionMetric?.error("Session terminated due to error");
+        } else if (this.serviceConfiguration.deli.checkServiceSummaryStatus && !this.serviceSummaryGenerated) {
```

Comment:
If `closeType === LambdaCloseType.Close` (indicates a regular process shutdown), then this shouldn't generate an error. 

So `LambdaCloseType.Close` should likely be part of the "session paused" case.

Reply:
Oh, I am assuming you are referring to LambdaCloseType.Stop. For that case, it is already being handled as session paused. In the case of the Stop enum, it would be going to the else block.

The line highlighted is to check if there was a successful service summary before the end of the session.

Reply:
Yea I meant `Stop`. I see it would be handled as pause. However it's the last branch in this if else chain. It's possible for the `this.serviceConfiguration.deli.checkServiceSummaryStatus && !this.serviceSummaryGenerated` branch to be taken when the lambda is being stopped and it would result in an error being reported, when it really shouldn't be

Reply:
Ah I see what you mean. Made the change.